indico/modules/rb_new/client/js/common/bookings/BookingDetailsModal.jsx

```diff
@@ -160,33 +171,186 @@ class BookingDetailsModal extends React.Component {
-            )}
-            {canReject && (
-                <Button type="button"
-                        icon="remove circle"
-                        color="red"
-                        size="small"
-                        content={<Translate>Reject booking</Translate>} />
-            )}
-            {showAccept && (
-                <Button type="button"
-                        icon="check circle"
-                        color="green"
-                        size="small"
-                        content={<Translate>Accept booking</Translate>} />
-            )}
-        </Modal.Actions>
-    );
+    renderActionButtons = (canCancel, canReject, showAccept) => {
+        const {bookingStateChangeInProgress} = this.props;
+        const {actionInProgress, openConfirm} = this.state;
+        const rejectButton = (
+            <Button type="button"
+                    icon="remove circle"
+                    color="red"
+                    size="small"
+                    loading={actionInProgress === 'reject' && bookingStateChangeInProgress}
+                    disabled={bookingStateChangeInProgress}
+                    content={<Translate>Reject booking</Translate>} />
+        );
+
+        const renderForm = ({handleSubmit, hasValidationErrors, submitSucceeded, submitting, pristine}) => {
+            return (
+                <Form styleName="rejection-form" onSubmit={handleSubmit}>
+                    <Field name="reason"
+                           component={ReduxFormField}
+                           as={TextArea}
+                           format={formatters.trim}
+                           placeholder={Translate.string('Provide the rejection reason')}
+                           rows={2}
+                           validate={validators.required}
+                           disabled={submitting}
+                           required
+                           formatOnBlur />
+                    <Button type="submit"
+                            disabled={submitting || pristine || hasValidationErrors || submitSucceeded}
+                            loading={submitting}
+                            floated="right"
+                            primary>
+                        <Translate>Reject</Translate>
+                    </Button>
+                </Form>
+            );
+        };
+
+        return (
+            <Modal.Actions>
+                {canCancel && (
+                    <>
+                        <Button type="button"
+                                icon="cancel"
+                                size="small"
+                                onClick={() => this.showConfirm('cancel')}
+                                loading={actionInProgress === 'cancel' && bookingStateChangeInProgress}
+                                disabled={bookingStateChangeInProgress}
+                                content={<Translate>Cancel booking</Translate>} />
```

Comment:
Does `Translate.string()` not work here?  most other props below use that one

Reply:
It should. Don't know why I did it like that here.